### PR TITLE
Add Date in days conversion support

### LINKED ISSUE #167: Dates handled improperly on load
After speaking with @randyzwitch  we found that `DATE` types are handled in mapd 4.4 as epoch days rather than epoch seconds. After loading data via pymapd, dates can be found to be invalid:

image

Expected Behavior:
- Dates show up when selected

Observed Behavior:
- Dates do not show up -- invalid date shows up

Reproduction:

Create table in OmniSci:
```
CREATE TABLE sales (
saleDate DATE
)
```

load dates
```
import pandas as pd
import pymapd

mapdcon = pymapd.connect(user=mapduser, password=mapdpass, host=mapdhost, dbname=mapddbname, port=mapdport, protocol=mapdprotocol)

# Create some dates
dates = pd.date_range('20190101', periods=10)
df = pd.DataFrame(dates)

# load into mapd 
mapdcon.load_table_columnar('sales', df, preserve_index=False)
```

### PR DESCRIPTION
**Pull request by wamsiv**
@andrewseidl Need to tag docker core build with latest commits in it, to enable the commented tests, they are failing on current build: `v4.5.0dev-20190104-5aee66d0e5`

fixes #167

### CODE REVIEW

File: pymapd/_pandas_loaders.py [hunk 1]
```diff
@@ -131,7 +132,8 @@ def build_input_columnar(df, preserve_index=True,
 
             if mapd_type in {'TIME', 'TIMESTAMP', 'DATE', 'BOOL'}:
                 # requires a cast to integer
-                data = thrift_cast(data, mapd_type)
+                print(col_types[colindex][1])
```
Comment (randyzwitch) [inline pymapd/_pandas_loaders.py]: Shouldn't need a print() statement here

**Comment by randyzwitch** [comment]
@wamsiv Since this issue started with OmniSci 4.4, why wouldn't we expect this functionality pass on any build?

File: pymapd/_pandas_loaders.py (hunk 1 above)
Comment (wamsiv) [reply pymapd/_pandas_loaders.py]: Forgot to remove it! Thanks!

**Comment by wamsiv** [comment]
Made some encoded level changes for small date. You can test the change against 4.5 and the latest master. Should see the difference in results.

**Comment by wamsiv** [comment]
We started exposing encoding to end points recently through a commit which is not available in this build. Without that there is not no way of knowing which type is being seeked.

**Comment by randyzwitch** [comment]
I guess what I'm asking is if the tests require a certain version of OmniSci, do we need to add some sort of OmniSci version logic into pymapd either in `connect()` or into the table loading code? It feels like we're breaking backwards compatibility unnecessarily, instead of fixing something that the user shouldn't need to worry about.

**Comment by wamsiv** [comment]
Looks like latest build container is failing at the start. And we are not breaking any compatibility, here I have added: https://github.com/omnisci/pymapd/pull/168/files#diff-51b57568ea5bfd539fbcf96d15ad4b18R86 another condition for date encoding in days. The legacy dates will still pass through `date_to_seconds(data)`.

**Comment by wamsiv** [comment]
@randyzwitch Ok I think I understand what you are confused about here. So, now the new columns which are getting created have encoding in days. If you try to use load_table here on new dates it would expect the input in days, which my logic does here but problem is that we were not exposing the encoding of new DATE type before. https://github.com/omnisci/mapd-core/commit/99adcec8b25d103cf1d58ac3d20c792b12fa8c30 now exposes the end point, which is not available in `4.4.2` hence when loading the data through `load_table_columnar` since we do not receive proper encoding for the type it assumes that encoding is NONE and proceeds to insert value in seconds (legacy way) rather than using new days logic.

**Comment by randyzwitch** [comment]
Ok, so it's 4.4.2 that unintentionally broke the compatibility by not exposing the endpoint? If so, can't do anything about that now.

This is failing because the newest version for Docker is still 4.4.2. @andrewseidl, if you could ping us when 4.5 is available, then we can merge this PR and get a new version of pymapd out the door

**Comment by wamsiv** [comment]
`v4.5.0 ` is available as the latest build and runs fine on local. Looks like this is a system issue, not with our container.

**Comment by randyzwitch** [comment]
LGTM, passes on my local machine. Needed to run `docker run --ipc=host -d -p 9091:6274 mapd/core-os-cpu` due to port remapping (per #163)

**Pull request merged**